### ISSUE TITLE
[Feature] Support ActivityPub

### ISSUE THREAD

**Issue by jkhgvfgvsth** [wintermeyer/vutuv#657]
**Feature**: Add [ActivityPub](https://www.wikipedia.org/wiki/ActivityPub) Support

**Why?**: To enable users to use their own domain for Vutuv and possibly self-host if Vutuv ever became discontinued for some reason. 

__________________________

Want to support this issue? Post a [bounty on it](https://www.bountysource.com/issues/76834913-feature-support-activitypub)! We accept bounties via [BountySource](https://www.wikipedia.org/wiki/Bountysource).

[![Bountysource](https://api.bountysource.com/badge/issue?issue_id=76834913)](https://www.bountysource.com/issues/76834913-feature-support-activitypub)

**Comment by wintermeyer** [comment]
Thoughts:
- We have bigger fish to fry right now. 
- It is an interesting idea. 

I will keep this issue open for anybody in the community who wants to tackle it. But if there is no interest I’ll close it. But I probably revisit it in the future to re-evaluate it.

**Comment by saschafoerster** [comment]
I am just waiting for this! I think Vutuv would fill a gap in the world of federated social networks: business networks!

**Comment by linux-lukas** [comment]
I would love this support!

**Comment by Rik44444** [comment]
same here, i arrived at this thread looking for a federated alternative to LinkedIn

**Comment by saschafoerster** [comment]
I think this could get big real quick, especially in the times, where federated alternatives become popular.

**Comment by linux-lukas** [comment]
I can only agree with @saschafoerster  and @Rik44444 . It would make vutuv a tremendously good open source and free alternative to XING and/or LinkedIn.

**Comment by aschrijver** [comment]
Quite frequent pops up in Fediverse discussion treads with people requesting a federated LinkedIn. Like [here recently](https://mastodon.xyz/@dougwade/110694450678797600). I dropped the link to Flockingbird then, which originally got NLnet funding for that idea (later they pivoted to become a job search bot). Check out the fediverse-related lists on https://delightful.club and you'll find a number of Elixir projects you can bootstrap from.

**Comment by beedell-roke** [comment]
@wintermeyer, this would be a way to acquire NLNet funding, and considering that [`Haui1112/FediWork`](https://github.com/Haui1112/FediWork) demonstrates that it's possible, Vutuv's potential value proposition — that it hosts an official server — is not dissimilar to `mastodon.social`'s.

**Comment by wintermeyer** [comment]
@RokeJulianLockhart I gave up the idea of any kind of funding for an open-source project many years ago. But I'll have a look into this issue.

**Comment by saschafoerster** [comment]
I am very happy to see, that this ticket has been re-opened. There is since a long time the wish for a good LinkedIn replacement in the Fediverse, even though some would see, there is not just need for "alternatives to" something, but new ideas are also very welcome. 
But for the space of "business fediverse", I would like to mention another initiative from another developper, that recently startet: https://codeberg.org/ProFed/ProFed
I think it is good to have multiple choices for different approaches to a professional network. But still it would be great of a maximum of interoperability could be achieved.

**Comment by aschrijver** [comment]
At this point in time there is also an ActivityPub library in Elixir, maintained by Bonfire Networks: https://github.com/bonfire-networks/activity_pub

**Comment by wintermeyer** [comment]
Assuming I am a boomer with very little technical know-how: How would a ActivityPub connectivity benefit both sides? What are advantages and what are disadvantages? Enlighten me please.

**Comment by beedell-roke** [comment]
@wintermeyer, you can both access each others' content, thereby nullifying the network effect that non-federated platforms utilise as their core advantage. Without federation, few non-FOSS developers nor hirers shall bother to utilise a platform like Vutuv, because no others are.

**Comment by wintermeyer** [comment]
> [@wintermeyer](https://github.com/wintermeyer), you can both access each others' content, thereby nullifying the network effect that non-federated platforms utilise as their core advantage. Without federation, few non-FOSS developers nor hirers shall bother to utilise a platform like Vutuv, because no others are.

@RokeJulianLockhart Do Facebook, X or LinkedIn offer this feature?

**Comment by aschrijver** [comment]
I don't use LinkedIn anymore (being active on the feed), but I keep my account for job searches where it is really useful (due to network effects, but also one-click apply for jobs based on your CV profile), and the professional 'Rolodex' is useful. Not just for the mutual connections you make, but also to check background on people I e-meet on the fediverse or on chat. That too benefits a lot from the network effects.

Vutuv becoming federated would allow first of all, all the self-hosted Vutuv installations to connect into a big social graph, and depending on features might allow admins or users to curate specialty groups. And it might federate with the several other linkedin alternatives that are in the works, given they all stick to an interoperable wire format. Also integrations with other app types can be made. For instance with microblogging, where Mastodon is a popular app. You could then send a job application as a microblog post. Right now things like that happen on the fediverse based on hashtag e.g. `#GetFediHired` when searching a job, or `#fossjobs` when offering (introduced by [@fossjobs](https://floss.social/@fossjobs)).

**Comment by beedell-roke** [comment]
@wintermeyer, no. If they did, they would lose their network effect. For established, unfederated platforms, it is their value proposition. That is why, when Facebook created Threads, they implemented AP federation, and continue to support it; to attract users. Threads was not established, so they required it, for it to become popular.

@aschrijver, indeed: whereas Vutuv would manage the profile composition, metadata, and connection aspects, the general AP network would provide the initial influx of posts that LinkedIn's profile post section, and feed, currently do.

**Comment by wintermeyer** [comment]
Superseded by #784, which rewrites this request as a clearer, self-contained issue: it states the real motivation (federation / network effects, not self-hosting), scopes what ActivityPub support concretely means for vutuv, weighs benefits vs. costs, and gathers the prior art and resources raised in this thread (the Elixir `activity_pub` library, FediWork, ProFed, Flockingbird).

Closing here to keep discussion in one place — please continue in #784. This thread stays as the original 2019 request and its community discussion. Thanks to everyone who weighed in here.

**Comment by beedell-roke** [comment]
@wintermeyer, perhaps, re-close this as a duplicate of https://github.com/wintermeyer/vutuv/issues/784, then? It being complete is inaccurate, and the duplicate pill directs users to the new issue, too.

**Comment by wintermeyer** [comment]
> [@wintermeyer](https://github.com/wintermeyer), perhaps, re-close this as a duplicate of [#784](https://github.com/wintermeyer/vutuv/issues/784), then? It being complete is inaccurate, and the duplicate pill directs users to the new issue, too.

It is closed.

**Comment by beedell-roke** [comment]
@wintermeyer, GitHub has added closure states (likely, since you were last active). You can see them in the context menu, beside the closure button. “Not Planned / Invalid” and “Duplicate” are the available alternatives to “Completed”. When one closes as an issue as duplicate, they are able to nominate another issue, by pasting its URI into the relevant form, as its successor.
